### PR TITLE
ix_updated: new semantics, provide unified old/new value has

### PR DESCRIPTION
**Pull request by rjbs**
As discussed, I think this interface will end up being easier in the future.

### CODE REVIEW

File: lib/Ix/DBIC/ResultSet.pm [hunk 1]
```diff
@@ -698,16 +698,18 @@ sub ix_update ($self, $ctx, $to_update) {
...
-        my %changed = $row->get_dirty_columns;
-        return $SKIPPED unless %changed;
+        my %new = $row->get_dirty_columns;
+        return $SKIPPED unless %new;
 
         $row->update({ modSeqChanged => $next_state });
 
-        # Hand the hook the old values for any columns that changed
-        my %orig = map {; $_ => $old{$_} } keys %changed;
+        if (%new && (my $code = $rclass->can('ix_updated'))) {
```
Comment (wolfsage) [inline lib/Ix/DBIC/ResultSet.pm]: `if (%new &&` is unnecessary at this point

Comment (wolfsage) [reply lib/Ix/DBIC/ResultSet.pm]: Also no reason to take `my $code` if we aren't using it

**Review by wolfsage — CHANGES_REQUESTED** [review]
2 nits, otherwise +1

File: lib/Ix/DBIC/ResultSet.pm (hunk 1 above)
Comment (rjbs) [reply lib/Ix/DBIC/ResultSet.pm]: Thanks, I'll fix these.

**Comment by rjbs** [comment]
I believe this addresses the nits.  Will have to coordinate this rollout.

**Comment by rjbs** [comment]
Merged, rolled out.

**Pull request merged**